src/main/asciidoc/driver/configuration.adoc

```diff
@@ -189,3 +195,54 @@ include::{python-examples}/test_examples.py[tag=tls-signed]
 ----
 --
 ====
+
+
+=== Authenticate user
+
+While the encrypted communication is established between a Neo4j driver and the server, the server may also require the driver to provide authentication credentials for the user connecting to the database.
+Authentication credentials are required if the server is configured to have authentication enabled.
+// See 'Securing Neo4j Server' (http://neo4j.com/docs/3.0.0-SNAPSHOT/security-server.html) for more information.
+
+[.tabbed-example]
+.Example: Connecting the driver to the server with authentication credentials
+====
+[include-with-dotnet]
+--
+[source, csharp]
+----
+include::{dotnet-examples}/Examples.cs[tags=connect-with-basic-auth]
+----
+--
+
+[include-with-java]
+--
+[source, java]
+----
+include::{java-examples}/Examples.java[tags=connect-with-basic-auth]
+----
+--
+[include-with-javascript]
+--
+[source, javascript]
+----
+include::{javascript-examples}/examples.test.js[tags=connect-with-basic-auth]
+----
+--
+[include-with-python]
+--
+[source, python]
+----
+include::{python-examples}/test_examples.py[tag=connect-with-basic-auth]
+----
+--
+====
+
+If communication between driver and server are is not encrypted, authentication credentials are sent as plain text.
+It is highly recommended to use authentication with encryption.
+
+[NOTE]
+--
+When connecting to Neo4j for the first time, the user must change the default password.
+Given a fresh installed Neo4j server, if a driver tries to connect to the Neo4j server with the default password, the attempt would probably fail with an error saying that the credentials are expired and a password reset is required.
```

Comment:
well, what I want to convey here is that the user has to change the default password via browser before they use a driver, otherwise they would got that error, as the for security reasons, the server should not allow the user to continue using the default password after the user logged in the first time with the default one. However the driver right now does not provide the interface for a user to change the password directly. So the user could only solve the problem by logging in via browser to change the default password. Please use your beautiful words to give this message to the user :)